bench/rttanalysis: TestBenchmarkExpectation failed [unexpectedly no result routers]

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16977303?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16977303?buildTab=artifacts#/) on master @ [c43628ced71b1089367e302f9aba0417baa6f43f](https://github.com/cockroachdb/cockroach/commits/c43628ced71b1089367e302f9aba0417baa6f43f):


```
    validate_benchmark_data.go:125: success: expected ORMQueries/hasura_column_descriptions to perform KV lookups in [5, 5], got 6
    validate_benchmark_data.go:125: success: expected AlterTableAddColumn/alter_table_add_1_column to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:125: success: expected Revoke/revoke_all_on_1_table to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:125: success: expected ORMQueries/has_sequence_privilege to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:125: success: expected Revoke/revoke_all_on_2_tables to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:125: success: expected AlterTableAddColumn/alter_table_add_2_columns to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:125: success: expected DropRole/drop_1_role to perform KV lookups in [34, 34], got 34
    validate_benchmark_data.go:125: success: expected Jobs/show_job to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:125: success: expected ORMQueries/hasura_column_descriptions_8_tables to perform KV lookups in [20, 20], got 20
    validate_benchmark_data.go:125: success: expected Jobs/pause_job to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:125: success: expected GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected Jobs/cancel_job to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:125: success: expected ORMQueries/activerecord_type_introspection_query to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:125: success: expected ORMQueries/hasura_column_descriptions_modified to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected Audit/select_from_an_audit_table to perform KV lookups in [5, 5], got 6
    validate_benchmark_data.go:125: success: expected GenerateObjects/generate_50000_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected Jobs/jobs_page_type_filtered to perform KV lookups in [3, 3], got 4
    validate_benchmark_data.go:125: success: expected DropRole/drop_2_roles to perform KV lookups in [43, 43], got 43
    validate_benchmark_data.go:125: success: expected Jobs/show_jobs to perform KV lookups in [3, 5], got 3
    validate_benchmark_data.go:125: success: expected ORMQueries/liquibase_migrations to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:122: fail: expected Jobs/jobs_page_latest_50 to perform KV lookups in [3, 3], got 5
    validate_benchmark_data.go:125: success: expected ORMQueries/pg_my_temp_schema_multiple_times to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:125: success: expected GenerateObjects/generate_100_tables_from_template to perform KV lookups in [11, 11], got 12
    validate_benchmark_data.go:125: success: expected GenerateObjects/generate_10_tables to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected ORMQueries/pg_attribute to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:125: success: expected AlterTableAddCheckConstraint/alter_table_add_3_check_constraints to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:125: success: expected AlterTableAddCheckConstraint/alter_table_add_2_check_constraints to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:125: success: expected ORMQueries/prisma_types_4 to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected AlterTableAddCheckConstraint/alter_table_add_1_check_constraint to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:125: success: expected ORMQueries/prisma_column_descriptions_updated to perform KV lookups in [3, 3], got 4
    validate_benchmark_data.go:125: success: expected ORMQueries/prisma_types_16 to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:125: success: expected Jobs/non_admin_show_jobs to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:125: success: expected Jobs/jobs_page_default to perform KV lookups in [3, 5], got 3
    validate_benchmark_data.go:125: success: expected Jobs/resume_job to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:125: success: expected ORMQueries/column_descriptions_json_agg to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:125: success: expected ORMQueries/liquibase_migrations_on_multiple_dbs to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:125: success: expected ORMQueries/prisma_column_descriptions to perform KV lookups in [133, 133], got 133
    validate_benchmark_data.go:125: success: expected ORMQueries/asyncpg_types to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:78: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:79: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:112: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation2728311517
--- FAIL: TestBenchmarkExpectation (190.64s)
--- FAIL: TestBenchmarkExpectation (191.19s)
=== RUN   TestBenchmarkExpectation/Jobs
    --- FAIL: TestBenchmarkExpectation/Jobs (141.76s)
=== RUN   TestBenchmarkExpectation/Jobs/jobs_page_latest_50
    rtt_analysis_bench.go:215: Jobs/jobs_page_latest_50: got 5, expected 3
    rtt_analysis_bench.go:221: wrote jaeger trace to /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation2728311517/Jobs_jobs_page_latest_50.jaeger.json
        --- FAIL: TestBenchmarkExpectation/Jobs/jobs_page_latest_50 (123.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126750 bench/rttanalysis: TestBenchmarkExpectation failed [Jobs/jobs_page_type_filtered_no_matches] [A-jobs C-test-failure O-robot P-2 T-jobs branch-release-24.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42392

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17878399?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17878399?buildTab=artifacts#/) on master @ [5023ee70c42f3d660df53f3c6d03b4c3ab2bc30f](https://github.com/cockroachdb/cockroach/commits/5023ee70c42f3d660df53f3c6d03b4c3ab2bc30f):


```
=== RUN   TestBenchmarkExpectation/AlterTableSplit
=== RUN   TestBenchmarkExpectation/DropDatabase/drop_database_2_tables
=== RUN   TestBenchmarkExpectation/ORMQueries/hasura_column_descriptions_8_tables
=== RUN   TestBenchmarkExpectation/ORMQueries/hasura_column_descriptions_modified
=== RUN   TestBenchmarkExpectation/ORMQueries/introspection_description_join
panic: unexpectedly no result routers in {0x402b3b6380 [] [] {[]} 0 local} [recovered]
	panic: unexpectedly no result routers in {0x402b3b6380 [] [] {[]} 0 local}

goroutine 420506 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x115?, {0x651b5f8, 0x409ae1adc0})
	pkg/util/stop/stopper.go:226 +0x68
panic({0x4a43460?, 0x410a8e6d70?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).GetResultTypes(0x40b1f0fe68?)
	pkg/sql/physicalplan/physical_plan.go:221 +0xb0
github.com/cockroachdb/cockroach/pkg/sql/physicalplan.(*PhysicalPlan).EnsureSingleStreamOnGateway(0x4056a22e80, {0x651b630, 0x410ad06990})
	pkg/sql/physicalplan/physical_plan.go:489 +0x70
github.com/cockroachdb/cockroach/pkg/sql.finalizePlanWithRowCount({0x651b630, 0x410ad06990}, 0x0?, 0x4056a22e80, 0x9b78b60?)
	pkg/sql/distsql_physical_planner.go:5122 +0x50
github.com/cockroachdb/cockroach/pkg/sql.FinalizePlan(...)
	pkg/sql/distsql_physical_planner.go:5109
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport.func1({0x651b630, 0x410ad06990}, 0x4058a40c60)
	pkg/sql/importer/import_processor_planning.go:111 +0x490
github.com/cockroachdb/cockroach/pkg/sql/importer.distImport({_, _}, {_, _}, _, _, {_, _, _}, {_, ...}, ...)
	pkg/sql/importer/import_processor_planning.go:115 +0x20c
github.com/cockroachdb/cockroach/pkg/sql/importer.ingestWithRetry({_, _}, {_, _}, _, _, {_, _, _}, {_, ...}, ...)
	pkg/sql/importer/import_job.go:1310 +0x738
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0x409ae1ac80, {0x651b630, 0x40f1e70ba0}, {0x4c644a0?, 0x40e9bd5200})
	pkg/sql/importer/import_job.go:310 +0x85c
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0x40b6b9f040?, 0x0?, 0x40b1f11868, {0x6514ce0?, 0x409ae1ac80?}, {0x651b630?, 0x40f1e70ba0?}, {0x4c644a0?, 0x40e9bd5200?})
	pkg/jobs/registry.go:1638 +0xf4
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0x40e82a6e08, {0x651b630, 0x40f1e70ae0}, {0x4c644a0, 0x40e9bd5200}, {0x6514ce0, 0x409ae1ac80}, 0x40b6b9f040, {0x40ee027819, 0x7}, ...)
	pkg/jobs/registry.go:1639 +0x880
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0x40e82a6e08, {0x651b630, 0x40f1e70ae0}, {0x6514ce0, 0x409ae1ac80}, 0x40b6b9f040, {0x40ee027819, 0x7}, {0x40f1e467c8, 0x8})
	pkg/jobs/adopt.go:392 +0x3e4
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x651b5f8?, 0x409ae1adc0?})
	pkg/jobs/adopt.go:245 +0xf0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x40f1e467b8?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x198
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 420498
	pkg/util/stop/stopper.go:488 +0x374
=== RUN   TestBenchmarkExpectation/Truncate/truncate_1_column_1_row
=== RUN   TestBenchmarkExpectation/VirtualTableQueries/select_crdb_internal.invalid_objects_with_1_fk
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_1_foreign_key
=== RUN   TestBenchmarkExpectation/RevokeRole/revoke_2_roles
=== RUN   TestBenchmarkExpectation/ShowGrants/grant_3_roles
=== RUN   TestBenchmarkExpectation/CreateRole
=== RUN   TestBenchmarkExpectation/DropRole/drop_3_roles
=== RUN   TestBenchmarkExpectation/GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables
=== RUN   TestBenchmarkExpectation/ORMQueries/column_descriptions_json_agg
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126750 bench/rttanalysis: TestBenchmarkExpectation failed [Jobs/jobs_page_type_filtered_no_matches] [A-jobs C-test-failure O-robot P-2 T-jobs branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/bench/rttanalysis/BUILD.bazel
+++ b/pkg/bench/rttanalysis/BUILD.bazel
@@ -57,6 +57,7 @@ go_test(
     exec_properties = {"test.Pool": "large"},
     deps = [
         "//pkg/base",
+        "//pkg/jobs",
         "//pkg/jobs/jobspb",
         "//pkg/security/securityassets",
         "//pkg/security/securitytest",

--- a/pkg/bench/rttanalysis/validate_benchmark_data_test.go
+++ b/pkg/bench/rttanalysis/validate_benchmark_data_test.go
@@ -5,6 +5,14 @@
 
 package rttanalysis
 
-import "testing"
+import (
+	"testing"
 
-func TestBenchmarkExpectation(t *testing.T) { reg.RunExpectations(t) }
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+)
+
+func TestBenchmarkExpectation(t *testing.T) {
+	defer jobs.TestingSetIDsToIgnore(map[jobspb.JobID]struct{}{3001: {}, 3002: {}})()
+	reg.RunExpectations(t)
+}

--- a/pkg/jobs/adopt.go
+++ b/pkg/jobs/adopt.go
@@ -143,6 +143,10 @@ func (r *Registry) processClaimedJobs(ctx context.Context, s sqlliveness.Session
 	for ok, err = it.Next(ctx); ok; ok, err = it.Next(ctx) {
 		row := it.Cur()
 		id := jobspb.JobID(*row[0].(*tree.DInt))
+		if _, skip := testingIgnoreIDs[id]; skip {
+			log.Warningf(ctx, "skipping execution of job %d as it is ignored by testing knob", id)
+			continue
+		}
 		claimedToResume[id] = struct{}{}
 	}
 	if err != nil {
@@ -151,6 +155,16 @@ func (r *Registry) processClaimedJobs(ctx context.Context, s sqlliveness.Session
 	r.filterAlreadyRunningAndCancelFromPreviousSessions(ctx, s, claimedToResume)
 	r.resumeClaimedJobs(ctx, s, claimedToResume)
 	return nil
+}
+
+var testingIgnoreIDs map[jobspb.JobID]struct{}
+
+// TestingSetIDsToIgnore is a test-only knob to set a list of job IDs that will
+// not be executed even after being claimed.
+func TestingSetIDsToIgnore(ignore map[jobspb.JobID]struct{}) func() {
+	orig := testingIgnoreIDs
+	testingIgnoreIDs = ignore
+	return func() { testingIgnoreIDs = orig }
 }
 
 // resumeClaimedJobs invokes r.resumeJob for each job in claimedToResume. It

